fastify.js

```diff
@@ -134,31 +131,33 @@ function build (options) {
     [pluginUtils.registeredPlugins]: [],
     // routes shorthand methods
     delete: function _delete (url, opts, handler) {
-      return prepareRoute.call(this, 'DELETE', url, opts, handler)
+      return router.prepareRoute.call(this, 'DELETE', url, opts, handler)
     },
     get: function _get (url, opts, handler) {
-      return prepareRoute.call(this, 'GET', url, opts, handler)
+      return router.prepareRoute.call(this, 'GET', url, opts, handler)
     },
     head: function _head (url, opts, handler) {
-      return prepareRoute.call(this, 'HEAD', url, opts, handler)
+      return router.prepareRoute.call(this, 'HEAD', url, opts, handler)
     },
     patch: function _patch (url, opts, handler) {
-      return prepareRoute.call(this, 'PATCH', url, opts, handler)
+      return router.prepareRoute.call(this, 'PATCH', url, opts, handler)
     },
     post: function _post (url, opts, handler) {
-      return prepareRoute.call(this, 'POST', url, opts, handler)
+      return router.prepareRoute.call(this, 'POST', url, opts, handler)
     },
     put: function _put (url, opts, handler) {
-      return prepareRoute.call(this, 'PUT', url, opts, handler)
+      return router.prepareRoute.call(this, 'PUT', url, opts, handler)
     },
     options: function _options (url, opts, handler) {
-      return prepareRoute.call(this, 'OPTIONS', url, opts, handler)
+      return router.prepareRoute.call(this, 'OPTIONS', url, opts, handler)
     },
     all: function _all (url, opts, handler) {
-      return prepareRoute.call(this, supportedMethods, url, opts, handler)
+      return router.prepareRoute.call(this, supportedMethods, url, opts, handler)
     },
     // extended route
-    route: route,
+    route: function _route (opts) {
+      return router.route.call(this, opts)
+    },
```

Comment:
Can you add a comment on why we are not using bind()? This pattern is repeated enough that I think it might be better to write an utility to generate these.

Reply:
This could fit?

> we need the fastify object that we are producing so we apply a lazy loading of the function otherwise we should bind it after the declaration

Reply:
Yes, that's ok.